bundle/config/mutator/load_git_details.go

```diff
@@ -21,45 +20,36 @@ func (m *loadGitDetails) Name() string {
 }
 
 func (m *loadGitDetails) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
-	// Load relevant git repository
-	repo, err := git.NewRepository(b.BundleRoot)
+	var diags diag.Diagnostics
+	info, err := git.FetchRepositoryInfo(ctx, b.BundleRoot.Native(), b.WorkspaceClient())
 	if err != nil {
-		return diag.FromErr(err)
+		diags = append(diags, diag.WarningFromErr(err)...)
 	}
 
-	// Read branch name of current checkout
-	branch, err := repo.CurrentBranch()
-	if err == nil {
-		b.Config.Bundle.Git.ActualBranch = branch
-		if b.Config.Bundle.Git.Branch == "" {
-			// Only load branch if there's no user defined value
-			b.Config.Bundle.Git.Inferred = true
-			b.Config.Bundle.Git.Branch = branch
-		}
-	} else {
-		log.Warnf(ctx, "failed to load current branch: %s", err)
+	b.WorktreeRoot = info.GuessedWorktreeRoot
```

Comment:
I understand it is from the comments and code for `FetchRepositoryInfo`, but can be confusing to read as often `err != nil` means the main return value is not populated or valid. Can you include a comment saying that it is actually valid?

Reply:
I agree. I thought it would be helpful, but looking back, it adds more confusion than value.

Removed this propery in https://github.com/databricks/cli//commit/1c3f5f2aa13877df9e998b66418cd3f9dcd42497